src/docfx/lib/JsonUtility.cs

```diff
@@ -135,6 +140,25 @@ public static (List<Error>, JToken) ValidateNullValue(this JToken token, JTokenS
             return (errors, token);
         }
 
+        public static List<Error> ValidateSchema(this JToken token, Type type, JTokenSourceMap mappings = null)
+        {
+            // TODO: Get the license somewhere
+            var registerErrors = RegisterLicense(string.Empty);
+            if (registerErrors.Any())
+                return registerErrors;
+
+            var generator = new JSchemaGenerator() { ContractResolver = new CamelCasePropertyNamesContractResolver() };
+            var schema = generator.Generate(type, true);
```

Comment:
>Generate [](start = 35, length = 8)

is this used reflection? can we cache the schema for the same type?

Reply:
Actually the more common way is to load the schema from template file. But sure, we can cache the schemas for types.

---
In reply to: [201892746](https://github.com/dotnet/docfx/pull/3052#discussion_r201892746) [](ancestors = 201892746)